### PR TITLE
p2p/pex: fix addrbook tests

### PR DESCRIPTION
**Pull request by gterzian**
Closes: #4458

## Description

______

For contributor use:

- [ ] Wrote tests
- [ ] Updated CHANGELOG_PENDING.md
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Updated relevant documentation (`docs/`) and code comments
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

**Comment by gterzian** [comment]
@erikgrinaker  Ready for review

**Comment by gterzian** [comment]
CI error appears unrelated

```
could not clone .git: unexpected client error: reference not found
Makefile:60: recipe for target 'proto-check-breaking' failed
```

**Comment by erikgrinaker** [comment]
Thanks! The proto failure is just flake, but there's an actual failure here as well:

```
--- FAIL: TestAddrBookSaveLoad (0.00s)
    addrbook_test.go:86: 
        	Error Trace:	addrbook_test.go:86
        	Error:      	Not equal: 
        	            	expected: 100
        	            	actual  : 0
        	Test:       	TestAddrBookSaveLoad
FAIL
FAIL	github.com/tendermint/tendermint/p2p/pex	7.803s
FAIL
```

I think you're just forgetting to call `.Start()` after creating the book.

**Comment by gterzian** [comment]
> I think you're just forgetting to call .Start() after creating the book.

Yep, that seems to have been the case, fixed it...

File: p2p/pex/addrbook_test.go [hunk 1]
```diff
@@ -93,19 +93,15 @@ func TestAddrBookLookup(t *testing.T) {
...
 	book.SetLogger(log.TestingLogger())
 	for _, addrSrc := range randAddrs {
 		addr := addrSrc.addr
 		src := addrSrc.src
 		book.AddAddress(addr, src)
 
-		ka := book.addrLookup[addr.ID]
-		assert.NotNil(t, ka, "Expected to find KnownAddress %v but wasn't there.", addr)
-
-		if !(ka.Addr.Equals(addr) && ka.Src.Equals(src)) {
```
Comment (melekes) [inline p2p/pex/addrbook_test.go]: what about this part?

Comment (gterzian) [reply p2p/pex/addrbook_test.go]: I couldn't find a way to test this using the current interface. I guess `HasAddress` could be changed to take an `src`?

**Pull request merged**